lib/src/signed_video_h26x_auth.c

```diff
@@ -309,16 +354,21 @@ verify_hashes_with_hash_list(signed_video_t *self, int *num_expected_nalus, int
       uint8_t *expected_hash = &expected_hashes[compare_idx * hash_size];
 
       if (memcmp(hash_to_verify, expected_hash, hash_size) == 0) {
-        // We have a match. Set validation_status and add missing nalus if we have detected any.
-        if (item->second_hash && !item->need_second_verification &&
-            item->nalu->is_first_nalu_in_gop) {
-          // If this |is_first_nalu_in_gop| it should be verified twice. If this the first time we
-          // signal that we |need_second_verification|.
-          DEBUG_LOG("This NALU needs a second verification");
-          item->need_second_verification = true;
+        if (!order_ok) {
```

Comment:
Here is an alternative solution which could work until I-frames are hashed twice. That is to set the first item of the GOP to `first_verification_not_authentic |= !order_ok;` before searching for matches in the list. The first item will then automatically be marked as 'N'.

I think it would be more robust if you somehow could mark the first item independently of the search for matching hashes, since any added if-statement means an extra "opportunity" to end up in a wrong place/state.

Reply:
In this case 
item->validation_status = item->first_verification_not_authentic |= !order_ok ? 'N' : '.';
would  other frames after I frames would be 'N' as well